src/go/k8s/pkg/resources/certmanager/type_helpers.go

```diff
@@ -0,0 +1,80 @@
+package certmanager
+
+import redpandav1alpha1 "github.com/vectorizedio/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+
+// Helper functions and types for Listeners
+
+// APIListener is a generic API Listener
+type APIListener interface {
+	// GetPort returns API port
+	GetPort() int
+	// GetTLS returns API TLSConfig
+	GetTLS() *redpandav1alpha1.TLSConfig
+	// GetExternal returns API's ExternalConnectivityConfig
+	GetExternal() *redpandav1alpha1.ExternalConnectivityConfig
+}
+
+func kafkaAPIListeners(r *redpandav1alpha1.Cluster) []APIListener {
+	listeners := []APIListener{}
+	for _, el := range r.Spec.Configuration.KafkaAPI {
+		listeners = append(listeners, el)
+	}
+	return listeners
+}
+
+func adminAPIListeners(r *redpandav1alpha1.Cluster) []APIListener {
+	listeners := []APIListener{}
+	for _, el := range r.Spec.Configuration.AdminAPI {
+		listeners = append(listeners, el)
+	}
+	return listeners
+}
+
+func schemaRegistryAPIListeners(r *redpandav1alpha1.Cluster) []APIListener {
+	if r.Spec.Configuration.SchemaRegistry == nil {
+		return []APIListener{}
+	}
+
+	return []APIListener{*r.Spec.Configuration.SchemaRegistry}
+}
+
+// PandaProxyAPIListeners returns all PandaProxyAPI listeners
+func pandaProxyAPIListeners(r *redpandav1alpha1.Cluster) []APIListener {
+	listeners := []APIListener{}
+	for _, el := range r.Spec.Configuration.PandaproxyAPI {
+		listeners = append(listeners, el)
+	}
+	return listeners
+}
+
+func getExternalTLSListener(listeners []APIListener) APIListener {
+	tlsListener := getTLSListener(listeners)
+	if tlsListener == nil {
+		return nil
+	}
+	if ext := tlsListener.GetExternal(); ext != nil && ext.Enabled {
+		return tlsListener
+	}
+	return nil
+}
```

Comment:
I don't understand why this function depends on the order of listeners. 

If user will create kafkAPI configuration as:

```yaml
spec:
  configuration:
    kafkaAPI:
    - external: enabled
      tls: enabled
      subdomain: "test"
    - port: 9999
```

The function will work, but if we reorder listeners:
```yaml
spec:
  configuration:
    kafkaAPI:
    - port: 9999
    - external: enabled
      tls: enabled
      subdomain: "test"   
```
then we will return nil. Am I missing something?

Reply:
Can you create unit tests to prove that using combination of 

|                        | getExternalTLSListener | getInternalTLSListener |
|------------------------|------------------------|------------------------|
| kafkaAPIListeners      |                       |                        |
| adminAPIListeners      |                        |                        |
| pandaProxyAPIListeners |                        |                        |

We can provide different order of listeners and the result will be always the same.